http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java

```diff
@@ -0,0 +1,176 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import static io.restassured.config.EncoderConfig.encoderConfig;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Marshaller;
+
+import org.apache.http.HttpStatus;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.OS;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class NonJsonPayLoadIT {
+    @QuarkusApplication(classes = { CityResource.class, City.class, CityListDTO.class, CityListWrapper.class,
+            CityListWrapperSerializer.class,
+    }, properties = "nonjson.properties")
+    static RestService app = new RestService();
+
+    private static final String EXPECTED_XML_PAYLOAD = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>" +
+            " <cityListDTO> <city> <name>Paris</name>" +
+            " <country>France</country>" +
+            " </city> </cityListDTO>";
+
+    private final static String EXPECTED_RESPONSE_BODY_YAML = "- description: \"Send a request with a YAML payload to the endpoint, including cities data\"\n"
+            +
+            "  steps:\n" +
+            "    - action: \"POST\"\n" +
+            "      endpoint: \"/media-type/yaml\"\n" +
+            "      requestHeaders:\n" +
+            "        Content-Type: \"application/yaml\"\n" +
+            "      requestBody:\n" +
+            "        cities:\n" +
+            "          - name: Tokio\n" +
+            "            country: Japan\n" +
+            "  expectedResponse:\n" +
+            "    status: 200\n" +
+            "    headers:\n" +
+            "      Content-Type: \"application/json\"\n" +
+            "    body:\n" +
+            "      cities:\n" +
+            "        - name: Tokio\n" +
+            "          country: Japan\n";
+
+    private final static String EXPECTED_RESPONSE_BODY_YAML_WINDOWS_OS = "- description: \"Send a request with a YAML payload to the endpoint, including cities data\"\r\n"
+            +
+            "  steps:\r\n" +
+            "    - action: \"POST\"\r\n" +
+            "      endpoint: \"/media-type/yaml\"\r\n" +
```

Comment:
this is hard to read, please use string block for all these long strings

Reply:
hmm, unless you need to backport it, but I don't think you do

Reply:
Yes, I'll use string block better.